src/renderer/src/lib/queries/happs.ts

```diff
@@ -41,10 +45,45 @@ export const createPublishersQuery = () => {
 	});
 };
 
-export const createAppStoreHappsQuery = () => {
+export const createAppVersionsQuery = () => (apphub_hrl: Uint8Array) => {
 	return createQuery({
-		queryKey: [APP_STORE_HAPPS_QUERY_KEY],
-		queryFn: () => getAppStoreClientOrThrow().appstoreZomeClient.getMyApps()
+		queryKey: [MY_HAPPS_ALL_VERSIONS_QUERY_KEY, apphub_hrl],
+		queryFn: async () => {
+			const devHubClient = getDevHubClientOrThrow();
+			const webappPackageEntryEntity =
+				await devHubClient.appHubZomeClient.getWebappPackageEntry(apphub_hrl);
+
+			const versions = await devHubClient.appHubZomeClient.getWebappPackageVersions(
+				webappPackageEntryEntity.id
+			);
+
+			return { versions, webapp_package_id: webappPackageEntryEntity.id };
+		}
+	});
+};
+
+export const createAppStoreMyHappsQuery = () => {
+	return createQuery({
+		queryKey: [APP_STORE_MY_HAPPS_QUERY_KEY],
+		queryFn: async () => {
+			const myApps = await getAppStoreClientOrThrow().appstoreZomeClient.getMyApps();
+			const appsWithIcons = await Promise.all(
+				myApps.map(async (app) => {
+					const icon = await getDevHubClientOrThrow().appHubMereMemoryZomeClient.getMereMemoryBytes(
+						app.content.icon
+					);
+
+					return {
+						id: uint8ArrayToURIComponent(app.id),
```

Comment:
As I understand, this conversion is here for the hash to be used in a URL? In that case I think it would be nice to use the `encodeHashToBase64` method of `@holochain/client` instead. Not very important but nice for consistency I think.

Reply:
Fixed in latest commit